### PR TITLE
Add number of datapoints logic

### LINKED ISSUE #979: Correct number of data points for fridge chart
Build Number: v2.2.0 rc-1?

Description: 

Currently the number of data points for the fridge chart is a static 30. In order to display a more pleasing chart when there are less data points, we need a more robust way to determine the number of data points.

Reproducible:

N/A

Reproduction Steps:

N/A

Comments:

Should use a 9 hour interval

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #979 

- Just simply uses either the `sensorLogs.length`, 30 or the quotient between the time now and the oldest `sensorLog` over 9 hours

### CODE REVIEW

File: src/utilities/modules/vaccines.js [hunk 1]
```diff
@@ -413,33 +415,40 @@ export const extractDataForBreachModal = ({
...
-
+  // Find the duration of of all sesorlogs and the current date. Leaves space
+  // in the chart indicating when the last sync cycle was.
+  const chartRangeMilliseconds = new Date().getTime() - sensorLogs.min('timestamp');
+  // Find the number of nine hour intervals during the duration of sensorlogs.
+  const nineHourIntervals = Math.floor(chartRangeMilliseconds / FRIDGE_CHART_DATA_POINT_INTERVAL);
+  // Use the maximum of either nine hour intervals, or the length of sensorlogs.
+  // If there are not enough sensorlogs to have a datapoint for each 9 hour interval,
+  // just use a datapoint for each sensorlog.
+  const chartIntervals = Math.max(nineHourIntervals, sensorLogs.length);
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/utilities/modules/vaccines.js]: Good point. I don't know what you mean by "data point count" though? Is that `MAX_FRIDGE_CHART_DATAPOINTS`?

Alternative could be to filter out all `breachAggregates` from `sensorLogs`

[the remark this answers is not in the document]
Comment (andreievg) [reply src/utilities/modules/vaccines.js]: @joshxg filtering breachAggregates would remove current ongoing breach from the graph. I was meaning, if sensorLog data is < FRIDGE_CHART_DATA_POINT_INTERVAL then use 'sensorLogs.length', how does that sound ?

[the remark this answers is not in the document]
Comment (joshxg) [reply src/utilities/modules/vaccines.js]: Didn't read properly, this was meant to be `Math.min` oops, but there is still a problem, when either there is such little data i.e. a couple of hours, there will be one data point. 

@andreievg so we don't use 9 hour intervals at all? Just use `const numberOfDataPoints = Math.min(sensorLogs.length, MAX_FRIDGE_CHART_DATA_POINTS)`? This is how `extractDataForBreachModal` works, I believe. I think this only works for that method because of the interval between logs is consistent.

Consider: 23 hours of data - 16 hours is aggregated, leaving ~2 full aggregation logs for the first 16 hours, and ~21 logs for the most recent ~7 hours (assuming no breaches). When `aggregateLogs` is run: we use 23 data points and will have gaps throughout the first 20 hours of logs (this is why my preference is to keep `preAggregate` logs for displaying charts as the consistent interval between logs  makes things a lot simpler).

Basically, if the number of `fullAggregationLogs < preAggregateLogs + breachAggregateLogs`, we get gaps (using length - consider having 10 `fullAggregateLogs` and 23 preAggregate logs, we use 30 data points, we would have gaps). This should be the case for the first ~8 days.

Unless, we use 9 hour intervals for this condition? However this does mean a limited graph for the first little while...


```
if (fullAggregationLogs.length < 0)
   dataPoints = 20minuteIntervals
else if(fullAggregationLogs.length < preAggregateLogs.length + breachAggregateLogs.length and also less than 30)
   dataPoints = 9 hour intervals
else
   dataPoints = 30
```

Hopefully this makes some sense..

[the remark this answers is not in the document]
Comment (andreievg) [reply src/utilities/modules/vaccines.js]: @joshxg 
That does kind of make sense. Can we simplify it further by:
* When doing full aggregation, don't delete preAggregates that are X days old ?
* When doing full aggregation start at preAggregates that are newer then last full aggregate
* When sensorLogs (with aggregation != '') are more then X days, use 9 hour intervals, otherwise 30min Intervals ?

**Comment by joshxg** [comment]
@andreievg Have added the changes, combined with the full aggregator changes I think this is what we are looking for
